pkg/k8s/service_cache.go

```diff
@@ -548,23 +548,27 @@ func (s *ServiceCache) mergeServiceUpdateLocked(service *serviceStore.ClusterSer
 		s.externalEndpoints[id] = externalEndpoints
 	}
 
-	scopedLog.Debugf("Updating backends to %+v", service.Backends)
-	backends := map[string]*Backend{}
-	for ipString, portConfig := range service.Backends {
-		backends[ipString] = &Backend{Ports: portConfig}
-	}
-	externalEndpoints.endpoints[service.Cluster] = &Endpoints{
-		Backends: backends,
+	if !service.Shared {
+		delete(externalEndpoints.endpoints, service.Cluster)
+	} else {
+		scopedLog.Debugf("Updating backends to %+v", service.Backends)
+		backends := map[string]*Backend{}
+		for ipString, portConfig := range service.Backends {
+			backends[ipString] = &Backend{Ports: portConfig}
+		}
+		externalEndpoints.endpoints[service.Cluster] = &Endpoints{
+			Backends: backends,
+		}
 	}
 
 	svc, ok := s.services[id]
 
 	endpoints, serviceReady := s.correlateEndpoints(id)
 
-	// Only send event notification if service is shared and ready.
+	// Only send event notification if service is ready.
 	// External endpoints are still tracked but correlation will not happen
 	// until the service is marked as shared.
-	if ok && svc.Shared && serviceReady {
+	if ok && serviceReady {
```

Comment:
remove check for shared here, as svc object here is local service, we want to publish event in case shared flag is toggled.